### PR TITLE
support encoding certificate policies in CertificateBuilder

### PR DESCRIPTION
**Pull request by reaperhulk**
I had this one mostly written already so I finished it. :smile:

Depends on #2231

### CODE REVIEW

**Comment by alex** [comment]
This is sufficiently complicated that I think the review is going to be a few passes, would you mind putting the bindings in their own PR so they're out of the way?

**Comment by reaperhulk** [comment]
depends on #2227

**Comment by alex** [comment]
Can be rebased

File: src/cryptography/hazmat/backends/openssl/backend.py [hunk 1]
```diff
@@ -98,6 +98,20 @@ def _encode_asn1_str(backend, data, length):
...
+    """
+    Create an ASN1_UTF8STRING from a Python unicode string.
+    This object will be a ASN1_STRING with UTF8 type in OpenSSL and
+    can be decoded with ASN1_STRING_to_UTF8.
+    """
+    s = backend._lib.ASN1_UTF8STRING_new()
+    res = backend._lib.ASN1_STRING_set(
+        s, string.encode("utf8"), len(string.encode("utf8"))
+    )
+    assert res == 1
```
Comment (alex) [inline src/cryptography/hazmat/backends/openssl/backend.py]: `openssl_assert`

Comment (alex) [inline src/cryptography/hazmat/backends/openssl/backend.py]: (Applies to this entire PR really)

File: src/cryptography/hazmat/backends/openssl/backend.py [hunk 2]
```diff
@@ -487,6 +576,7 @@ def _encode_crl_distribution_points(backend, crl_distribution_points):
     ExtensionOID.ISSUER_ALTERNATIVE_NAME: _encode_alt_name,
     ExtensionOID.EXTENDED_KEY_USAGE: _encode_extended_key_usage,
     ExtensionOID.AUTHORITY_KEY_IDENTIFIER: _encode_authority_key_identifier,
+    x509.OID_CERTIFICATE_POLICIES: _encode_certificate_policies,
```
Comment (alex) [inline src/cryptography/hazmat/backends/openssl/backend.py]: This should use `ExtensionOID`

**Pull request merged**